annotation-processor/src/main/java/com/hubspot/annotation/processor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,151 @@
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+import javax.tools.FileObject;
+import javax.tools.StandardLocation;
+
+import com.google.auto.service.AutoService;
+
+@SupportedAnnotationTypes("java.lang.SuppressWarnings")
+@AutoService(Processor.class)
+public class ModernizerAnnotationProcessor extends AbstractProcessor {
+
+    @Override
+    public final SourceVersion getSupportedSourceVersion() {
+        return SourceVersion.latest();
+    }
+
+    @Override
+    public final boolean process(
+        Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+        for (TypeElement annotation : annotations) {
+            List<String> annotatedClasses =
+                getAnnotatedClasses(roundEnv, annotation);
+            makeAnnotatedClassesFile(annotatedClasses);
+        }
+        return true;
+    }
+
+    private List<String> getAnnotatedClasses(
+        RoundEnvironment roundEnv, TypeElement annotation) {
+        List<String> annotatedClasses = new ArrayList<String>();
+        for (Element element : roundEnv.getElementsAnnotatedWith(annotation)) {
+            List<String> warnings = Arrays.asList(
+                    element.getAnnotation(SuppressWarnings.class).value());
+            if (warnings.contains("modernizer")) {
+                if (element.getKind().isClass()) {
+                    annotatedClasses.add(getClassHeader(element));
+                }
+            }
+        }
+        return annotatedClasses;
+    }
+
+    private void makeAnnotatedClassesFile(List<String> annotatedClasses) {
+        FileWriter writer = null;
+        try {
+            File dir = new File(getFilePath(), "modernizer");
+            if (!dir.exists()) {
+                dir.mkdir();
+            }
+            File file = new File(dir.getPath(),
+                "modernizer-ignore-annotated-classes.txt");
+            writer = new FileWriter(file);
+            for (String element : annotatedClasses) {
+                writer.write(element + "\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (writer != null) {
+                try {
+                    writer.close();
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+    }
+
+    private String getFilePath() {
+        try {
+            FileObject fileObjectToGetPath =
+                processingEnv.getFiler().createResource(
+                    StandardLocation.CLASS_OUTPUT,
+                    "",
+                    "dummy-file.txt",
+                    null);
+            String path = fileObjectToGetPath.getName();
+            fileObjectToGetPath.delete();
+            while (!path.endsWith("target")) {
```

Comment:
This will definitely break for non-standard output directories. Maybe we only walk up if we know that it ends with `target/classes` or `target/test-classes`. In that case we could also just do a simple substring.

Reply:
I think this works for non-standard output directories which are inside `target` directory. In those cases, since we are walking up until target and then creating the file, it shouldn't break. May be we can walk up until we find `target` if we find that the path contains `/target/`? We could check that using a regex instead of just checking for `/target/classes` or `/target/test-classes`. Please correct me if I'm wrong.

Reply:
I agree with what you're saying, but I think that this is a pretty complex edgecase that it could be best to punt on for now. It's not really a blocker for our particular use case and we can come back to it in a subsequent PR.